loopd/swapclient_server.go

```diff
@@ -494,8 +494,13 @@ func (s *swapClientServer) GetLoopInQuote(ctx context.Context,
 
 		lastHop = &lastHopVertex
 	}
-
-	routeHints, err := unmarshallRouteHints(req.LoopInRouteHints)
+	var routeHints [][]zpay32.HopHint
```

Comment:
probably want some validation here that the client doesn't set both? Doesn't really make sense for caller to set both

Reply:
Generally my preference is to keep validation in on the server side so that all calls are subject to the same validation. If we just keep the validation in cli then we might miss adding it if we add a new call to this server endpoint. 

In terms of validation here, can't you just have: 
```
haveRouteHints := len(req.LoopInRequestHints) !=0

switch{
    case haveRouteHints && req.Private: 
         return "err can't set both"
         
     case haveRouteHints: 
        routeHints, err = unmarshallRouteHints(req.LoopInRouteHints)
        
     case: req.Private: 
         routeHints, err = loop.SelectHopHints(ctx, s.lnd,...
}
```

> however here I'm not it's possible to tell if a the private flag is set or not set since it should default to false

I believe that the only option that the client has is `--private` in our flags library, they can't set `--private=false` so we don't need to worry about default values.